lib/src/json_schema/validator.dart

```diff
@@ -134,9 +134,9 @@ class Validator {
   Set<Uri> _vocabulary;
 
   /// Keep track of the number of evaluated items contexts in a list, treating the list as a stack.
-  /// The context is an [int], representing the number of successful evaluations for the list in the
+  /// The context is an [List] of [bool], representing the number of successful evaluations for the list in the
   /// given context.
-  List<int> _evaluatedItemsContext = [];
+  List<List<bool>> _evaluatedItemsContext = [];
```

Comment:
Draft 2020 make it explicit that we need to be tracking the position of items that have been validated.